pkg/metrics/timer/timer.go

```diff
@@ -0,0 +1,123 @@
+package timer
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// Timer is the struct that keeps track of each of the sections.
+type Timer struct {
+	listOfStages []string
+	stageTimes   map[string]time.Duration
+	startTimes   map[string]time.Time
+}
+
+const (
+	// InfrastructureStageName is a constant string value that is used to display the Infrastructure created stage text in the logs.
+	InfrastructureStageName = "Infrastructure"
+	// APIStageName is a constant string value that is used to display the API stage text in the logs.
+	APIStageName = "API"
+	// BootstrapCompleteStageName is a constant string value that is used to display the Bootstrap Complete text in the logs.
+	BootstrapCompleteStageName = "Bootstrap Complete"
+	// BootstrapDestroyStageName is a constant string value that is used to display the Boostrap Destroy Complete text in the logs.
+	BootstrapDestroyStageName = "Bootstrap Destroy"
+	// ClusterOperatorStageName is a constant string value that is used to display the Cluster Operator Complete text in the logs.
+	ClusterOperatorStageName = "Cluster Operator"
+
+	// ConsoleStageName is a constant string value that is used to display the Cluster Operator Complete text in the logs.
+	ConsoleStageName = "Console"
+
+	// TotalTimeElapsed is a constant string value to denote total time elapsed.
+	TotalTimeElapsed = "Total"
+)
+
+var timer = Timer{
+	listOfStages: []string{},
+	stageTimes:   make(map[string]time.Duration),
+	startTimes:   make(map[string]time.Time),
+}
+
+// StartTimer initiailzes the timer object with the current timestamp information.
+func StartTimer(key string) {
```

Comment:
Can we reuse the functions with the `Timer` pointer receivers so we don't have two copies of all the functions? Can we use `NewTimer` to make the global timer?

Reply:
> Can we reuse the functions with the Timer pointer receivers so we don't have two copies of all the functions? Can we use NewTimer to make the global timer?

Actually I have changed these and committed but not pushed yet. Waiting on Abhinav's comment on what the input should be for the Summarize function should look like and I'll push them together.